build_tools/cmake/build_android.sh

```diff
@@ -30,30 +33,55 @@ ROOT_DIR=$(git rev-parse --show-toplevel)
 
 CMAKE_BIN=${CMAKE_BIN:-$(which cmake)}
 
-"$CMAKE_BIN" --version
+"${CMAKE_BIN?}" --version
 ninja --version
 
 cd ${ROOT_DIR?}
-if [ -d "build" ]
+
+# --------------------------------------------------------------------------- #
+# Build for the host.
+
+if [ -d "build-host" ]
 then
-  echo "Build directory already exists. Will use cached results there."
+  echo "build-host directory already exists. Will use cached results there."
 else
-  echo "Build directory does not already exist. Creating a new one."
-  mkdir build
+  echo "build-host directory does not already exist. Creating a new one."
+  mkdir build-host
 fi
+cd build-host
+
+# Configure, build, install.
+"${CMAKE_BIN?}" -G Ninja .. \
+  -DCMAKE_INSTALL_PREFIX=./install \
+  -DIREE_BUILD_COMPILER=ON \
+  -DIREE_BUILD_TESTS=OFF \
+  -DIREE_BUILD_SAMPLES=OFF
+"${CMAKE_BIN?}" --build .
+"${CMAKE_BIN?}" --install .
```

Comment:
This `--install` isn't doing what I expect it to on our BuildKite CI ([logs](https://buildkite.com/iree/iree-android-arm64-v8a/builds/3488#f3d82754-c278-4f1f-8e08-8550eac7157d))

```
[01:20:31 UTC] /usr/bin/cmake --version
  | cmake version 3.13.5

...

[2992/2992] Linking CXX executable iree/tools/iree-run-mlir
  | [[01:29:59 UTC] /usr/bin/cmake --install .
  | -- Building HAL drivers: DyLib;VMLA;Vulkan
  | -- Building target backends: DYLIB-LLVM-AOT;Metal-SPIRV;Vulkan-SPIRV;VMLA
  | -- Adding bundled LLVM source dependency
...
  | -- SPIRV-Cross: Finding Git version for SPIRV-Cross.
  | -- SPIRV-Cross: Git hash: 685f864
  | -- Configuring done
  | -- Generating done
  | -- Build files have been written to: /usr/src/github/iree/build-host
  | [[01:30:08 UTC] cd /usr/src/github/iree
```

https://stackoverflow.com/a/34040747 points out this was added in 3.15, so `cmake --build . --target install` should be used on earlier versions.